### PR TITLE
Added PlayerMock#simulateInventoryClick

### PR DESCRIPTION
**Pull request by PeachesMLG**
# Description
Added PlayerMock#simulateInventoryClick

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Unit tests added.
- [x] Code follows existing code format.

# Info on creating a pull request
- Make sure that unit tests are added which test the relevant changes.
- Make sure that the changes follow the existing code format.

# For maintainer
When a PR is approved, the maintainer should label this PR with `release/*`.

- If the PR fixes a bug in MockBukkit, update the patch version. (if the current version is `0.4.1`, the new version should be `0.4.2`)
- If the PR adds a new feature to MockBukkit, update minor version. (if the current version is `0.4.1`, the new version should be `0.5.0`)

Note that a PR that fixes an `UnimplementedOperationException` should be considered a new version and not a bugfix.

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 1]
```diff
@@ -256,7 +259,7 @@ protected BlockDamageEvent simulateBlockDamagePure(Block block)
 	public @Nullable BlockBreakEvent simulateBlockBreak(Block block)
 	{
 		if ((gamemode == GameMode.SPECTATOR || gamemode == GameMode.ADVENTURE)
```
Comment (thelooter) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java:261]: Please fix indents

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 2]
```diff
@@ -1266,12 +1307,12 @@ public void chat(@NotNull String msg)
...
-		    Bukkit.getOnlinePlayers().stream().map(p -> (Audience) p).collect(Collectors.toSet()),
-		    ChatRenderer.defaultRenderer(),
-		    Component.text(msg),
-		    Component.text(msg)
+				true,
+				this,
+				Bukkit.getOnlinePlayers().stream().map(p -> (Audience) p).collect(Collectors.toSet()),
+				ChatRenderer.defaultRenderer(),
+				Component.text(msg),
+				Component.text(msg)
```
Comment (thelooter) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java:1315]: Please fix indents

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 3]
```diff
@@ -1371,27 +1412,27 @@ public void playNote(@NotNull Location loc, @NotNull Instrument instrument, @Not
...
+					case IRON_XYLOPHONE -> Sound.BLOCK_NOTE_BLOCK_IRON_XYLOPHONE;
+					case PIANO -> Sound.BLOCK_NOTE_BLOCK_HARP;
+					case PLING -> Sound.BLOCK_NOTE_BLOCK_PLING;
+					case SNARE_DRUM -> Sound.BLOCK_NOTE_BLOCK_SNARE;
+					case STICKS -> Sound.BLOCK_NOTE_BLOCK_HAT;
+					case XYLOPHONE -> Sound.BLOCK_NOTE_BLOCK_XYLOPHONE;
+					default ->
+						// This should never be reached unless Mojang adds new instruments
+							throw new UnimplementedOperationException("Instrument '" + instrument + "' has no implementation!");
+				};
```
Comment (thelooter) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java:1435]: Please fix indents

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 4]
```diff
@@ -2396,60 +2435,60 @@ public <T> void spawnParticle(@NotNull Particle particle, double x, double y, do
 
 	@Override
 	public void spawnParticle(@NotNull Particle particle, @NotNull Location location, int count, double offsetX, double offsetY,
-	                          double offsetZ)
+							  double offsetZ)
```
Comment (thelooter) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java:2438]: Please fix indents

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 5]
```diff
@@ -2396,60 +2435,60 @@ public <T> void spawnParticle(@NotNull Particle particle, double x, double y, do
...
 	{
 		this.spawnParticle(particle, location.getX(), location.getY(), location.getZ(), count, offsetX, offsetY, offsetZ);
 	}
 
 	@Override
 	public void spawnParticle(@NotNull Particle particle, double x, double y, double z, int count, double offsetX,
-	                          double offsetY, double offsetZ)
+							  double offsetY, double offsetZ)
```
Comment (thelooter) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java:2445]: Please fix indents

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 6]
```diff
@@ -2396,60 +2435,60 @@ public <T> void spawnParticle(@NotNull Particle particle, double x, double y, do
...
 	{
 		this.spawnParticle(particle, x, y, z, count, offsetX, offsetY, offsetZ, null);
 
 	}
 
 	@Override
 	public <T> void spawnParticle(@NotNull Particle particle, @NotNull Location location, int count, double offsetX, double offsetY,
-	                              double offsetZ, T data)
+								  double offsetZ, T data)
```
Comment (thelooter) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java:2453]: Please fix indents

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 7]
```diff
@@ -2396,60 +2435,60 @@ public <T> void spawnParticle(@NotNull Particle particle, double x, double y, do
...
 	{
 		this.spawnParticle(particle, location.getX(), location.getY(), location.getZ(), count, offsetX, offsetY, offsetZ, data);
 	}
 
 	@Override
 	public <T> void spawnParticle(@NotNull Particle particle, double x, double y, double z, int count, double offsetX,
-	                              double offsetY, double offsetZ, T data)
+								  double offsetY, double offsetZ, T data)
```
Comment (thelooter) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java:2460]: Please fix indents

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 8]
```diff
@@ -2396,60 +2435,60 @@ public <T> void spawnParticle(@NotNull Particle particle, double x, double y, do
...
 	{
 		this.spawnParticle(particle, x, y, z, count, offsetX, offsetY, offsetZ, 1, data);
 
 	}
 
 	@Override
 	public void spawnParticle(@NotNull Particle particle, @NotNull Location location, int count, double offsetX, double offsetY,
-	                          double offsetZ, double extra)
+							  double offsetZ, double extra)
```
Comment (thelooter) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java:2468]: Please fix indents

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 9]
```diff
@@ -2396,60 +2435,60 @@ public <T> void spawnParticle(@NotNull Particle particle, double x, double y, do
...
 	{
 		this.spawnParticle(particle, location.getX(), location.getY(), location.getZ(), count, offsetX, offsetY, offsetZ, extra);
 	}
 
 	@Override
 	public void spawnParticle(@NotNull Particle particle, double x, double y, double z, int count, double offsetX,
-	                          double offsetY, double offsetZ, double extra)
+							  double offsetY, double offsetZ, double extra)
```
Comment (thelooter) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java:2475]: Please fix indents

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 10]
```diff
@@ -2396,60 +2435,60 @@ public <T> void spawnParticle(@NotNull Particle particle, double x, double y, do
...
 	{
 		this.spawnParticle(particle, x, y, z, count, offsetX, offsetY, offsetZ, extra, null);
 
 	}
 
 	@Override
 	public <T> void spawnParticle(@NotNull Particle particle, @NotNull Location location, int count, double offsetX, double offsetY,
-	                              double offsetZ, double extra, T data)
+								  double offsetZ, double extra, T data)
```
Comment (thelooter) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java:2483]: Please fix indents

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 11]
```diff
@@ -2396,60 +2435,60 @@ public <T> void spawnParticle(@NotNull Particle particle, double x, double y, do
...
 	{
 		this.spawnParticle(particle, location.getX(), location.getY(), location.getZ(), count, offsetX, offsetY, offsetZ, data);
 
 	}
 
 	@Override
 	public <T> void spawnParticle(@NotNull Particle particle, double x, double y, double z, int count, double offsetX,
-	                              double offsetY, double offsetZ, double extra, T data)
+								  double offsetY, double offsetZ, double extra, T data)
```
Comment (thelooter) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java:2491]: Please fix indents

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 12]
```diff
@@ -2978,7 +3017,7 @@ public void sendMultiBlockChange(@NotNull Map<Location, BlockData> blockChanges,
 
 	@Override
 	public void sendEquipmentChange(@NotNull LivingEntity entity, @NotNull EquipmentSlot slot,
-	                                @NotNull ItemStack item)
+									@NotNull ItemStack item)
```
Comment (thelooter) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java:3020]: Please fix indents

File: src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java [hunk 13]
```diff
@@ -81,12 +83,13 @@
...
 	// Taken from https://minecraft.gamepedia.com/Experience#Leveling_up
 	private static int[] expRequired =
-	{
-		7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37, 42, 47, 52, 57, 62, 67, 72, 77, 82, 87, 92, 97, 102,
-		107, 112, 121, 130, 139, 148, 157, 166, 175, 184, 193
-	};
+			{
+					7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37, 42, 47, 52, 57, 62, 67, 72, 77, 82, 87, 92, 97, 102,
+					107, 112, 121, 130, 139, 148, 157, 166, 175, 184, 193
+			};
```
Comment (thelooter) [inline src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java:92]: Please fix indents

File: src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java [hunk 14]
```diff
@@ -1006,7 +1009,7 @@ void testPlaySound()
 		player.assertSoundHeard(sound, audio ->
 		{
 			return player.getLocation().equals(audio.getLocation()) && audio.getCategory() == SoundCategory.AMBIENT
-			&& audio.getVolume() == volume && audio.getPitch() == pitch;
+					&& audio.getVolume() == volume && audio.getPitch() == pitch;
```
Comment (thelooter) [inline src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java:1012]: Please fix indents

File: src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java [hunk 15]
```diff
@@ -1021,7 +1024,7 @@ void testPlaySoundString()
 		player.assertSoundHeard(sound, audio ->
 		{
 			return player.getEyeLocation().equals(audio.getLocation()) && audio.getCategory() == SoundCategory.RECORDS
-			&& audio.getVolume() == volume && audio.getPitch() == pitch;
+					&& audio.getVolume() == volume && audio.getPitch() == pitch;
```
Comment (thelooter) [inline src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java:1027]: Please fix indents

File: src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java [hunk 16]
```diff
@@ -1033,7 +1036,7 @@ void testPlayNote_NewMethod()
 		player.assertSoundHeard(Sound.BLOCK_NOTE_BLOCK_BANJO, audio ->
 		{
 			return player.getEyeLocation().equals(audio.getLocation()) && audio.getCategory() == SoundCategory.RECORDS
-			&& audio.getVolume() == 3.0f && Math.abs(audio.getPitch() - Math.pow(2.0D, (note - 12.0D) / 12.0D)) < 0.01;
+					&& audio.getVolume() == 3.0f && Math.abs(audio.getPitch() - Math.pow(2.0D, (note - 12.0D) / 12.0D)) < 0.01;
```
Comment (thelooter) [inline src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java:1039]: Please fix indents

File: src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java [hunk 17]
```diff
@@ -1045,7 +1048,7 @@ void testPlayNote_OldMethod()
 		player.assertSoundHeard(Sound.BLOCK_NOTE_BLOCK_HARP, audio ->
 		{
 			return player.getEyeLocation().equals(audio.getLocation()) && audio.getCategory() == SoundCategory.RECORDS
-			&& audio.getVolume() == 3.0f && Math.abs(audio.getPitch() - Math.pow(2.0D, (note - 12.0D) / 12.0D)) < 0.01;
+					&& audio.getVolume() == 3.0f && Math.abs(audio.getPitch() - Math.pow(2.0D, (note - 12.0D) / 12.0D)) < 0.01;
```
Comment (thelooter) [inline src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java:1051]: Please fix indents

File: src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java [hunk 18]
```diff
@@ -1057,7 +1060,7 @@ void testCloseInventoryEvenFired()
 		player.setItemOnCursor(new ItemStack(Material.PUMPKIN));
 		player.closeInventory();
 		server.getPluginManager().assertEventFired(InventoryCloseEvent.class,
-		        e -> e.getPlayer() == player && e.getInventory() == inv);
+				e -> e.getPlayer() == player && e.getInventory() == inv);
```
Comment (thelooter) [inline src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java:1063]: Please fix indents

**Review by thelooter — CHANGES_REQUESTED** [review]

**Comment by PeachesMLG** [comment]
My IDE auto formatted it according to: https://github.com/MockBukkit/MockBukkit/blob/v1.18/.editorconfig tho.
If your using a different code style, u should probably change that file :shrug:

**Comment by thelooter** [comment]
Didn't see that I'm sorry. Seems correct then.

**Review by thelooter — APPROVED** [review]

**Pull request merged**